### PR TITLE
Update flow

### PR DESCRIPTION
**Pull request by nwoltman**
Also remove the caret `^` from the version in package.json because snapshot tests need the Flow version to be consistent (the yarn lockfile does take care of this, but it's nice to be explicit and also support npm users).

### CODE REVIEW

File: test/4.example.js [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import React, { Component } from 'react';
 
 type Props = {
-  firstName: Object,
+  firstName: number,
```
Comment (amilajack) [inline test/4.example.js]: Why was this changed from `Object`?

File: test/codebases/project-1/4.example.js [hunk 2]
```diff
@@ -4,7 +4,7 @@
 import React, { Component } from 'react';
 
 type Props = {
-  firstName: Object,
+  firstName: number,
```
Comment (amilajack) [inline test/codebases/project-1/4.example.js]: Why was this changed from `Object`?

File: test/4.example.js (hunk 1 above)
Comment (nwoltman) [reply test/4.example.js]: Flow doesn’t treat the `Object` type the same way anymore. It acts like the `any` type now, so this change was needed so that passing a string as the `firstName` prop would continue to produce an error.

Also, Flow highly discourages using `Object` now (https://flow.org/en/docs/types/objects/#toc-object-type):

> **`Object` is unsafe and should be avoided.**

**Comment by amilajack** [comment]
Thanks for this! 🎉 🎉 🎉 I'll publish this with the next tag soon

**Comment by amilajack** [comment]
published to `eslint-plugin-flowtype-errors@next`. Please test this out and get back to me after some testing
